### DISCUSSION TITLE
document training like CodeGPT plus

### DISCUSSION THREAD

**Discussion by revbin** [danny-avila/LibreChat#850] [Feature Requests & Suggestions]
Hi

if its possible to implement document upload to pinecone like codegpt plus and enable desired uploaded document to specific agent or persona .

demo : https://plus.codegpt.co/

regards

**Comment by danny-avila** [comment]
I'll be focusing document integration/vectorization next.

**Comment by revbin** [comment]
most effective implementation to chat with doc i come a cross .the result  have  a high accuracy even with gpt-3.5 .  maybe give you some insight for future inplemation in Librechat .

demo : https://lightpdf.com/chatdoc

**Reply by danny-avila** [reply]
this is closed-source so not much I can learn from it

**Comment by mikekuzak** [comment]
+1 for document upload!

**Reply by danny-avila** [reply]
this is in active development!

**Comment by eburnette** [comment]
Also interested in this. Check out one open source implementation at https://github.com/pinecone-io/canopy. They get good results with several techniques, one is they ask gpt-3.5-turbo which queries are needed, so it can answer questions that require multiple lookups.

I think this would be really powerful with the new Claude-3 embedding model, but there are some good open-source ones too.
